tensorflow/compiler/xla/service/gpu/target_util.cc

```diff
@@ -167,7 +193,15 @@ llvm::CallInst* EmitCallToTargetIntrinsic(
   if (target_triple.isNVPTX()) {
     llvm_intrinsic_id = gpu_intrinsic_id.nvptx_intrinsic;
   } else if (target_triple.getArch() == llvm::Triple::amdgcn) {
```

Comment:
I think this is a good location to put your explanation as a comment, once @b-sumner can confirm. Something like below

> // On amdgpu target, these operations are made as device functions instead of intrinsics,
// and there's no plan to add these device functions as LLVM intrinsics. Therefore, we 
// have to invoke device functions in this subroutine instead.

Reply:
will do.

Reply:
I think it's better to comment at the declaration of `TargetIntrinsics`.

```
// Wrapper structure for carrying llvm intrinsic ids for NVPTX/AMDGPU platforms.
// On AMDGPU, some of these operations are made as device functions instead of
// intrinsics. Therefore a variant type is used to wrap the lambda to call
// those device functions.
struct TargetIntrinsics {
  llvm::Intrinsic::ID nvptx_intrinsic;
  absl::variant<llvm::Intrinsic::ID,
                std::function<llvm::CallInst*(llvm::IRBuilder<>*)>>
      amdgpu_intrinsic_or_function;
};
```

Reply:
Agreed